Deprecate the methods of `default.qubit` (#4145)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* `qml.grouping` module is removed. The functionality has been reorganized in the `qml.pauli` module.

* The public methods of `DefaultQubit` are pending changes to
  follow the new device API, as used in `DefaultQubit2`.

>   The docstrings of the affected methods in `DefaultQubit` have been
>   updated to reflect this.

Reviewer: ```suggestion
  Warnings have been added to the docstrings to reflect this.
```
